packages/generator/src/parser/edmx-to-vdm-v4.ts

```diff
@@ -0,0 +1,70 @@
+/* Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. */
+import { VdmNavigationProperty, VdmComplexType, VdmEntity } from '../vdm-types';
+import { ServiceNameFormatter } from '../service-name-formatter';
+import { EdmxEntityType, EdmxEntitySet } from './parser-types-v4';
+import { isCollection } from './parser-util';
+import {
+  JoinedEntityMetadata,
+  ParsedServiceMetadata
+} from './parser-types-common';
+import {
+  joinEntityMetadata,
+  createEntityClassNames,
+  transformEntity,
+  navigationPropertyBase
+} from './edmx-to-vdm-common';
+
+function navigationProperties(
+  entityMetadata: JoinedEntityMetadata,
+  classNames: { [originalName: string]: string },
+  formatter: ServiceNameFormatter
+): VdmNavigationProperty[] {
+  const entityType = entityMetadata.entityType as EdmxEntityType;
```

Comment:
Ah here is the cast now. I like it much better than the conditional type.

Reply:
cool 😊